translator/util/ec2util/ec2util.go

```diff
@@ -67,46 +76,54 @@ func initEC2UtilSingleton() (newInstance *ec2Util) {
 		log.Println("W! [EC2] Sleep until network is up")
 		time.Sleep(1 * time.Second)
 	}
+
 	if !networkUp {
 		log.Println("E! [EC2] No available network interface")
 	}
 
-	ses, err := session.NewSession()
+	err := newInstance.getEC2MetadataFromIMDS()
+
 	if err != nil {
-		log.Println("E! [EC2] getting new session info: ", err)
-		return
+		log.Println("E! [EC2] Cannot get EC2 Metadata from IMDS:", err)
 	}
-	md := ec2metadata.New(ses)
 
-	if !md.Available() {
-		log.Println("E! ec2metadata is not available")
-		return
-	}
+	return
+}
 
-	if info, err := md.GetMetadata("instance-id"); err == nil {
-		newInstance.InstanceID = info
-	} else {
-		log.Println("E! getting instance-id from EC2 metadata fail: ", err)
+func (e *ec2Util) getEC2MetadataFromIMDS() error {
+	ses, err := session.NewSession()
+
+	if err != nil {
+		return err
 	}
 
-	if info, err := md.GetMetadata("hostname"); err == nil {
-		newInstance.Hostname = info
-	} else {
-		log.Println("E! getting hostname from EC2 metadata fail: ", err)
+	md := ec2metadata.New(ses,
+		&aws.Config{
+			Retryer: client.DefaultRetryer{NumMaxRetries: allowedIMDSRetries},
```

Comment:
What is the frequency of retries/jitter for a `DefaultRetryer`? Is 2 retries sufficient?

Reply:
If the AWS SDK already changed the default behavior to 1 second timeout and 2 retries, why do we have to create our own retryer?

Reply:
👍  Someone can argue with this: mostly for visibility and also consistent with the `EC2Tagger Retryer`. I'm fine with deleting this custom Retryer so non-block from me

Reply:
Then I'll ask, why is the ec2tagger retryer being changed if they both use the same AWS SDK code, which based on the PR you linked, should both already be using the corrected default retry?

Reply:
No the existing EC2 Tagger Retried Strategies used [our configuration](https://github.com/aws/amazon-cloudwatch-agent/blob/master/cfg/aws/credentials.go#L94-L104) 
```
      config := &aws.Config{
		Region:                        aws.String(c.Region),
		CredentialsChainVerboseErrors: aws.Bool(true),
		HTTPClient:                    &http.Client{Timeout: 1 * time.Minute},
		LogLevel:                      SDKLogLevel(),
		Logger:                        SDKLogger{},
	}
```

Therefore, the IMDS from Translation uses the default one while EC2 Tagger Retried used the custom one.